src/implementations/twig/components/button/button.story.js

```diff
@@ -11,117 +9,162 @@ import dataCall from '@ecl/specs-component-button/demo/data--call';
 import dataGhost from '@ecl/specs-component-button/demo/data--ghost';
 import dataSearch from '@ecl/specs-component-button/demo/data--search';
 
+import defaultSprite from '@ecl/resources-ec-icons/dist/sprites/icons.svg';
 import button from './button.html.twig';
 import notes from './README.md';
 
 const iconsList = {};
-iconsList.none = '';
-
 uiIcons.forEach((icon) => {
   iconsList[icon] = icon;
 });
 
-// Preserve the adapted specs.
-const prepareButton = (data, type) => {
-  const typeLabel = type ? tabLabels.required : tabLabels.optional;
-  data.disabled = boolean('disabled', data.disabled, tabLabels.states);
-  data.label = text('label', data.label, tabLabels.required);
-
-  data.variant = select(
-    'variant',
-    [data.variant],
-    data.variant,
-    tabLabels.required
-  );
-
-  data.type = select('type', [data.type], data.type, typeLabel);
-
-  getExtraKnobs(data);
+const prepareControls = (data, args) => {
+  data.label = args.label;
+  data.disabled = args.disabled;
```

Comment:
We can try having Object.assign() merge before the latter specifics and if/else statements.

Reply:
Not sure to fully understand here, if you ever have a more precise example, or update the PR... this will help me.

Reply:
Just as an example, although not a problem not having it. In this case, the `data` is the target: `data = Object.assign(data, args)`

```javascript
var data = { label: 'specsLabel', prop: 'specProp'}
var args = { label: 'argsLabel', prop: 'argProp'}

Object.assign(data, args)

console.log(data) // {label: "argsLabel", prop: "argProp"}
```